airbyte-server/src/main/java/io/airbyte/server/ServerApp.java

```diff
@@ -233,6 +234,14 @@ public static ServerRunnable getServer(final ServerFactory apiFactory,
     final Flyway jobsFlyway = FlywayFactory.create(jobsDataSource, DbMigrationHandler.class.getSimpleName(), JobsDatabaseMigrator.DB_IDENTIFIER,
         JobsDatabaseMigrator.MIGRATION_FILE_LOCATION);
 
+    // It is important that the migration to the temporal scheduler is performed before the server
```

Comment:
Is this true with the server being configured is HA? Won't we accept request while starting the new version of the server?

Reply:
@benmoriceau technically you are right, but our documentation instructs our open source users to first spin down their airbyte deployment before spinning it back up. If they don't follow this and try to do some sort of rolling deploy, then their request may still be sent to the old server pod (configured to use the old scheduler) while the new one is spinning up, but this should only be a temporary situation until the old server pod scales down, and any weird job state resulting from that should be corrected by the temporal workflow.

Reply:
Putting the migration logic in the ServerApp startup here is meant to prevent requests coming to the server that is _configured to use the new scheduler_ from being processed before the migration is complete, and only the new server pods even in a rolling deployment should be configured to use the new scheduler, so those are the pods that need to block on the migration